src/test/groovy/com/vacation_bot/vacation/VacationIntegrationTest.groovy

```diff
@@ -0,0 +1,177 @@
+package com.vacation_bot.vacation
+
+import com.vacation_bot.AbstractSpockIntegrationTest
+import com.vacation_bot.domain.models.UserModel
+import com.vacation_bot.domain.models.VacationRequestBody
+import com.vacation_bot.domain.models.VacationTotalModel
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import spock.lang.Shared
+import spock.lang.Unroll
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+
+/**
+ * Integration level test of {@Link VacationGateway}.
+ */
+class VacationIntegrationTest extends AbstractSpockIntegrationTest {
+
+    def restTemplate = new TestRestTemplate()
+
+    def headers = new HttpHeaders(accept: Arrays.asList(MediaType.APPLICATION_JSON))
+
+
+    @Shared
+    def startDate = '2017-10-02'
+
+    @Shared
+    def endDate = '2017-10-26'
+
+    @Shared
+    def static final DEFAULT_VACATION_TOTAL_DAYS = 20
+
+    static final String USER_ID = UUID.randomUUID()
+    def static final USER_NAME = 'Alex'
+    def static final ALIASES = ['Batman', 'Joker', 'Superman']
+
+    def inputData = new VacationRequestBody(userName: USER_NAME, startDate: startDate, endDate: endDate)
+
+    static final int CURRENT_YEAR = Calendar.getInstance().get(Calendar.YEAR)
+
+    @Unroll
+    def 'exercise vacation gateway (vacation total is null)'() {
```

Comment:
Read in the Spock documentation how you can variate the test description/name with the `@Unroll` annotation.